sdk/resources/mgmt/src/main/java/com/azure/management/resources/fluentcore/arm/collection/implementation/IndependentChildrenImpl.java

```diff
@@ -81,7 +79,7 @@ public T getById(String id) {
     @Override
     public Mono<T> getByIdAsync(String id) {
         ResourceId resourceId = ResourceId.fromString(id);
-        if (resourceId == null) {
+        if (resourceId.parent() == null) {
```

Comment:
why `parent()`? maybe only delete this line?

Reply:
Since the next line has `resourceId.parent().name()`, so I add `parent()` in it. Is it better not to check this NPE?